internal/storetest/testresult.go

```diff
@@ -68,83 +67,127 @@ func (result TestResult) FriendlyFailuresDisplay() string {
 	listObjectsResultsOutput := ""
 
 	if totalCheckCount > 0 {
-		for index := 0; index < totalCheckCount; index++ {
-			checkResult := result.CheckResults[index]
+		failedCheckCount, checkResultsOutput = buildCheckTestResults(
+			totalCheckCount, result, failedCheckCount, checkResultsOutput)
+	}
 
-			if !checkResult.IsPassing() {
-				failedCheckCount++
+	if totalListObjectsCount > 0 {
+		failedListObjectsCount, listObjectsResultsOutput = buildListObjectsTestResults(
+			totalListObjectsCount, result, failedListObjectsCount, listObjectsResultsOutput)
+	}
+
+	if failedCheckCount+failedListObjectsCount != 0 {
+		return buildTestResultOutput(
+			result, totalCheckCount,
+			failedCheckCount, totalListObjectsCount,
+			failedListObjectsCount, checkResultsOutput,
+			listObjectsResultsOutput)
+	}
 
-				got := "N/A"
-				if checkResult.Got != nil {
-					got = strconv.FormatBool(*checkResult.Got)
-				}
-
-				checkResultsOutput = fmt.Sprintf(
-					"%s\nⅹ Check(user=%s,relation=%s,object=%s, context=%v): expected=%t, got=%s, error=%v",
-					checkResultsOutput,
-					checkResult.Request.User,
-					checkResult.Request.Relation,
-					checkResult.Request.Object,
-					checkResult.Request.Context,
-					checkResult.Expected,
-					got,
-					checkResult.Error,
-				)
+	return ""
+}
+
+func buildCheckTestResults(totalCheckCount int,
```

Comment:
I extracted one function for each result (Check, ListObjects, Summary) to address cyclomatic complexity lint errors